alpha/apps/kaltura/lib/live/kLiveStreamConsumer.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+
+class kLiveStreamConsumer implements kObjectChangedEventConsumer
+{
+	/**
+	* @inheritDoc
+	 */
+	public function objectChanged(BaseObject $object, array $modifiedColumns)
+	{
+		if ($object instanceof LiveEntry && $this->isLiveEntryCategoryChanged($object, $modifiedColumns))
+		{
+			$this->handleLiveEntryCategoryChanged($object);
+			return true;
+		}
+		
+		if ($object instanceof categoryEntry)
+		{
+			$entryId = $object->getEntryId();
+			$entry = entryPeer::retrieveByPK($entryId);
+			if ($entry)
+			{
+				$this->handleLiveEntryCategoryChanged($entry);
+			}
+		}
+		
+		return true;
+	}
+
+	protected function handleLiveEntryCategoryChanged(LiveEntry $liveEntry)
+	{
+		$recordedEntryId = $liveEntry->getRecordedEntryId();
+		$recordedEntry = BaseentryPeer::retrieveByPK($recordedEntryId);
+		if (!$recordedEntry)
+		{
+			KalturaLog::info("Recorded entry $recordedEntryId could not be retrieved.");
+			return false;
+		}
+		
+		if (!PermissionPeer::isValidForPartner(PermissionName::FEATURE_DISABLE_CATEGORY_LIMIT, $liveEntry->getPartnerId()))
+		{
+			$categories = $liveEntry->getCategories();
+			if (is_null($categories))
+			{
+				KalturaLog::info("Categories for live entry {$liveEntry->getId()} could not be retrieved.");
+				return false;
+			}
+			
+			$recordedEntry->setCategories($categories);
+			$recordedEntry->save();
+		}
+		else
+		{
+			$this->compareAndSyncCategories($liveEntry, $recordedEntryId, $recordedEntry);
+		}
+		
+		return true;
+	}
+	
+	protected function compareAndSyncCategories($liveEntry, $recordedEntryId, $recordedEntry)
```

Comment:
you dont need  $recordedEntryId as param you can get it from $recordedEntry->getId